### ISSUE TITLE
1.20.1: Still Struggling with colorblind mode.

### ISSUE THREAD

**Issue by HorusApl** [kangarko/ChatControl#2386]
### Are you using MySQL?

No

### Are you using BungeeCord/Velocity?

No

### Question

Been working on this for far too long, and hoping someone can give me the answer. I have a variable called colorless that does a permission check if a player has hide.color, and then changes the color in my chat.yml format so that it can be per player. No matter if the permission is true or false, I just can't seem to get it correctly. 
[debug.zip](https://github.com/kangarko/ChatControl-Red/files/11798119/debug.zip)

**Comment by kangarko** [comment]
Can you navigate me in which format you use the variable?

**Comment by HorusApl** [comment]
In the chat.yml format is the {colorless} variable, and that variable is suitably named colorless.yml

**Comment by HorusApl** [comment]
Not sure if I'm even barking up the right tree. Just want to have so if player has X perm, that any chat messages show up as white for them.

**Comment by kangarko** [comment]
Oh then I would simply create a chat rule that would require a certain permission and then remove all colors in a message for a player. Can you try doing that?

**Comment by HorusApl** [comment]
That was my initial plan and I asked on the discord, I tried:

match (.*)
require perm hide.color
strip colors true
then replace $0

but it wasn't doing anything at all, and then I was directed to do it via the variable/format way which also doesn't seem to do it properly.

**Comment by kangarko** [comment]
Try setting Rules.Strip_Colors to true in settings.yml. Make sure you place this rule in rules/chat.rs. That works for me on my end, let me know how it goes.

**Comment by HorusApl** [comment]
I just tried this and while this works for manually typed color codes "&bHello!" will show as white. I make active use of /chc color menu, which this does nothing for and is the primary thing that I need.

**Comment by kangarko** [comment]
Then just split the format part containing the two variables and apply the Sender_Permission there such as:

![a](https://i.imgur.com/RmyUHGb.png)

**Comment by kangarko** [comment]
Such that the color-and-decoration have the variables and the message part only has the message

**Comment by HorusApl** [comment]
I have followed your exact format, and this just turns all messages white and shows no color in any messages. I gave the permission, and still no color. I took away the Send_Permission part, and still it's white. It seems unless {chat_color} and {chat_decoration} are with {message}, they don't function.

  color-and-decoration:
    Message: '&7: {chat_color}{chat_decoration}'
    Sender_Permission: can.have.color

  # And finally, show the message the player has typed out, prepending it with player's
  # suffix from your permission plugin. We replace {chat_color} and {chat_decoration} with the 
  # color and decoration set in '/chc color'.
  message:
    Message: '{message}'

**Comment by HorusApl** [comment]
![image](https://github.com/kangarko/ChatControl-Red/assets/64336395/f6b6435b-d63e-4fbe-a7fd-94498a85f427)

**Comment by HorusApl** [comment]
Alright this is my current format: https://mclo.gs/astp1l8

If I deny the permission for hide.simplechat, all the correct variables aren't appearing **Except** for the colors. The colors still show through even though the permission is negated. As seen in the screenshot, denying the permission does nothing to remove the color which is what is supposed to happen.
![image](https://github.com/kangarko/ChatControl-Red/assets/64336395/3bfa9ed8-f83a-47c6-b26b-9bbf535efc39)

**Comment by kangarko** [comment]
Change Receiver_Permission to Sender_Permission because right now it's evaluated for each chat message received, not yo as the sender.

**Comment by HorusApl** [comment]
But then that won't make all messages white/unformatted for that particular player, it would only make messages that they send as white/unformatted. So while that would work for an individual to be sending white messages, the purpose is so that if someone runs a command /colorblind, they will be given a permission and all messages in chat to them will appear as white.

**Comment by kangarko** [comment]
So sorry, I have spent time on this but found that due to how chat components are built, this is a non-workable issue.

Specifically when a color is inserted we need to split the word into a new chat "component", since this is can have performance penalty we save performance by not running receiver_permission conditions when we built the component. Instead, those are checked later when sending the each component, but the message has already been replaced. Which works fine for any other variable but specifically for the color and decoration one this splits the component.

I would need to scrape the chat renderer which I dont have enough time for, for now. I could use modern approach but it would require MC 1.16+ and we cant do that for backward compatibility reasons. Sorry for this, right now this specific placeholder appears to be a wontfix
